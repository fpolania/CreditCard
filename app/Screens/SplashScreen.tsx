import React, { useEffect } from 'react';
import { View, Text, StyleSheet, Image, StatusBar } from 'react-native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { useNavigation } from '@react-navigation/native';

/**
 * SplashScreen luego de 4000 remplza la url y nos muestra el home
 *
 * @return {*} 
 */
const SplashScreen = () => {
  const navigation = useNavigation();

  useEffect(() => {
    const timer = setTimeout(() => {
      navigation.navigate('Home' as never);
    }, 4000);

    return () => clearTimeout(timer);
  }, [navigation]);

  return (
    <View style={styles.container}>
      <StatusBar barStyle="light-content" backgroundColor="#4F6D7A" />
      <Image source={require('../../assets/images/partial-react-logo.png')} style={styles.logo} />
      <Text style={styles.title}>Bienvenido a Mi App</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#4F6D7A',
  },
  logo: {
    width: 150,
    height: 150,
    marginBottom: 20,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#FFFFFF',
  },
});

export default SplashScreen;
