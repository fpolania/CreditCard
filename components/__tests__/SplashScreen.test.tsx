import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import SplashScreen from '@/app/Screens/SplashScreen';

jest.mock('react-native/Libraries/Components/StatusBar/StatusBar', () => 'StatusBar');

describe('SplashScreen', () => {
  it('debe renderizar el componente correctamente', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <SplashScreen />
      </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('debe mostrar el texto de bienvenida', () => {
    const { queryByText } = render(
      <NavigationContainer>
        <SplashScreen />
      </NavigationContainer>
    );
    const welcomeText = queryByText('Bienvenido a Mi App');
    expect(welcomeText).not.toBeNull();
  });
});
