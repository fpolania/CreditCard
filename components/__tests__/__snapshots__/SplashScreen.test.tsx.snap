// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen debe renderizar el componente correctamente 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4F6D7A",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <StatusBar
    backgroundColor="#4F6D7A"
    barStyle="light-content"
  />
  <Image
    source={
      {
        "testUri": "../../../assets/images/partial-react-logo.png",
      }
    }
    style={
      {
        "height": 150,
        "marginBottom": 20,
        "width": 150,
      }
    }
  />
  <Text
    style={
      {
        "color": "#FFFFFF",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Bienvenido a Mi App
  </Text>
</View>
`;
